### PR TITLE
added homeWork3

### PR DESCRIPTION
**Pull request by lesiaturuta**
Acceptance Criteria for 1

- [x]  Mentor is added to pr
- [x]  AC copied to pr description
- [x]  Link to deployed version is added to pr
- [x]  The ProductsList component correctly receives product data as props.
- [x]  ProductCard component created and renders a single product.
- [x]  Header and Footer elements are present.
- [x]  ProductList displays right on mobile devices

Acceptance Criteria for 1*

- [x]  The Header component has clickable elements that toggle the display between the About and ProductsList components.
- [x]  State management is handled in App.tsx, with state determining which component to show.
- [x]  LocalStorage added to save items in card after page realod

link: https://lesiaturuta.github.io/fe-react-2024/

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,17 +1,66 @@
...
+    };
+
     return (
         <div className="body">
-            <HeaderComponent />
-            <AboutComponent />
+            <HeaderComponent amountCart={amountCart} nav={nav} />
+            {!isProduct && <AboutComponent />}
+            {isProduct && <ProductList products={products} increaseCounter={increaseCounter} decrementCounter={decrementCounter} />}
             <FooterComponent />
```
Comment (vadym-zinchenko-moc) [inline src/App.tsx:51]: Screenshot 2024-05-08 at 17 42 09

I think better to add margin. And for header too.

File: src/components/header/Header.component.tsx [hunk 2]
```diff
@@ -7,39 +9,79 @@ import sun from '@/assets/icons/sun.svg';
...
+            </nav>
+            <div className={styles.buttons}>
+                <div className={styles.cart}>
+                    <div
+                        className={styles.cart_button}
+                        style={{
+                            backgroundImage: `url(${cart})`,
+                        }}
+                    ></div>
+                    {amountCart && (
```
Comment (vadym-zinchenko-moc) [inline src/components/header/Header.component.tsx]: Screenshot 2024-05-08 at 17 32 11

I don't see icons

File: src/components/header/Header.component.tsx [hunk 3]
```diff
@@ -7,39 +9,79 @@ import sun from '@/assets/icons/sun.svg';
...
+                <div className={styles.verticalLine}></div>
+                <img src={moon} alt="moon" />
+            </div>
+            <nav className={styles.nav}>
+                <ul className={styles.nav__items}>
+                    <li className={`${styles.nav__item} ${styles.mr_49}`}>
+                        <input
+                            type="button"
+                            className={`${styles.nav__link} ${activeNavItem === 'About' && styles.nav__link__active}`}
+                            value="About"
```
Comment (vadym-zinchenko-moc) [inline src/components/header/Header.component.tsx]: Screenshot 2024-05-08 at 17 34 55

I am not sure that yellow color is suitable. And need to add cursor pointer for links i think.

File: src/App.tsx [hunk 4]
```diff
@@ -1,17 +1,66 @@
...
+            .then((data) => {
+                let count: number = 0;
+                setProducts(data);
+                data.forEach((product: Product) => {
+                    if (localStorage.getItem(product.id.toString()) === 'true') {
+                        count++;
+                    }
+                });
+                setAmountCart(count);
+            })
```
Comment (vadym-zinchenko-moc) [inline src/App.tsx:22]: i think you can create additional method for this. Too much logic in useEffect.

File: src/App.tsx [hunk 5]
```diff
@@ -1,17 +1,66 @@
...
+
+    const increaseCounter = () => {
+        setAmountCart(amountCart + 1);
+    };
+
+    const decrementCounter = () => {
+        setAmountCart(amountCart - 1);
+    };
+
+    const nav = (link: String) => {
```
Comment (vadym-zinchenko-moc) [inline src/App.tsx]: need to change to a more understandable name

File: src/App.tsx [hunk 6]
```diff
@@ -1,17 +1,66 @@
...
+                setIsProduct(false);
+                break;
+            }
+            case 'Products': {
+                setIsProduct(true);
+                break;
+            }
+            default: {
+                break;
+            }
```
Comment (vadym-zinchenko-moc) [inline src/App.tsx]: I think it can be done using the ternary operator more briefly

File: src/components/productCard/ProductCard.component.tsx [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+    const [isCart, setIsCart] = useState(false);
+
+    useEffect(() => {
+        const storedValue = localStorage.getItem(product.id.toString());
+        if (storedValue !== null) {
+            const isProductInCart = JSON.parse(storedValue);
+            setIsCart(isProductInCart);
+        }
+    }, [product.id]);
+    const changeCart = () => {
```
Comment (vadym-zinchenko-moc) [inline src/components/productCard/ProductCard.component.tsx:27]: add space between methods

File: src/components/productCard/ProductCard.component.tsx [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+        setIsCart(!isCart);
+        localStorage.setItem(product.id.toString(), JSON.stringify(!isCart));
+        if (!isCart) {
```
Comment (vadym-zinchenko-moc) [inline src/components/productCard/ProductCard.component.tsx]: use ternary operator. It more readable and short 

`isCart ? decrementCounter() : increaseCounter()`

**Review by vadym-zinchenko-moc — CHANGES_REQUESTED** [review]

File: src/App.tsx (hunk 1 above)
Comment (lesiaturuta) [reply src/App.tsx:51]: fixed

File: src/components/header/Header.component.tsx (hunk 2 above)
Comment (lesiaturuta) [reply src/components/header/Header.component.tsx]: fixed

File: src/components/header/Header.component.tsx (hunk 3 above)
Comment (lesiaturuta) [reply src/components/header/Header.component.tsx]: fixed

File: src/App.tsx (hunk 4 above)
Comment (lesiaturuta) [reply src/App.tsx:22]: Created new method

File: src/App.tsx (hunk 5 above)
Comment (lesiaturuta) [reply src/App.tsx]: fixed

File: src/App.tsx (hunk 6 above)
Comment (lesiaturuta) [reply src/App.tsx]: fixed

File: src/components/productCard/ProductCard.component.tsx (hunk 7 above)
Comment (lesiaturuta) [reply src/components/productCard/ProductCard.component.tsx:27]: fixed

File: src/components/productCard/ProductCard.component.tsx (hunk 8 above)
Comment (lesiaturuta) [reply src/components/productCard/ProductCard.component.tsx]: fixed

File: src/components/header/Header.component.tsx [hunk 9]
```diff
@@ -7,39 +9,69 @@ import sun from '@/assets/icons/sun.svg';
...
+            <img className={styles.mr_49} src={logo} alt="logo" />
+            <div className={styles.themes}>
+                <img src={sun} alt="sun" />
+                <div className={styles.verticalLine}></div>
+                <img src={moon} alt="moon" />
+            </div>
+            <nav className={styles.nav}>
+                <ul className={styles.nav__items}>
+                    <li className={`${styles.nav__item} ${styles.mr_49}`}>
+                        <input
```
Comment (vadym-zinchenko-moc) [inline src/components/header/Header.component.tsx]: why you don't use just `<button/>`?

File: src/App.tsx [hunk 10]
```diff
@@ -1,17 +1,57 @@
...
+        setAmountCart(amountCart - 1);
+    };
+
     return (
         <div className="body">
-            <HeaderComponent />
-            <AboutComponent />
+            <HeaderComponent
+                amountCart={amountCart}
+                nav={(link: String) => (link === 'About' ? setIsProduct(false) : setIsProduct(true))}
```
Comment (vadym-zinchenko-moc) [inline src/App.tsx]: Better to write it like method. I meant that name must be more understandable. example - `toggleCurrentPage`

**Review by vadym-zinchenko-moc — CHANGES_REQUESTED** [review]

File: src/components/header/Header.component.tsx (hunk 9 above)
Comment (lesiaturuta) [reply src/components/header/Header.component.tsx]: fixed

File: src/App.tsx (hunk 10 above)
Comment (lesiaturuta) [reply src/App.tsx]: ok. I created method toggleCurrentPage

**Review by vadym-zinchenko-moc — APPROVED** [review]

**Pull request merged**